sched/task/task_posixspawn.c

```diff
@@ -345,91 +269,27 @@ int posix_spawn(FAR pid_t *pid, FAR const char *path,
    * structure.
    */
 
-  /* Get exclusive access to the global parameter structure */
-
-  ret = nxmutex_lock(&g_spawn_parmlock);
-  if (ret < 0)
-    {
-      serr("ERROR: nxmutex_lock failed: %d\n", ret);
-      return -ret;
-    }
-
-  /* Populate the parameter structure */
-
-  g_spawn_parms.result       = ENOSYS;
-  g_spawn_parms.pid          = pid;
-  g_spawn_parms.file_actions = file_actions ? *file_actions : NULL;
-  g_spawn_parms.attr         = attr;
-  g_spawn_parms.argv         = argv;
-  g_spawn_parms.envp         = envp;
-  g_spawn_parms.u.posix.path = path;
-
-  /* Get the priority of this (parent) task */
-
-  ret = nxsched_get_param(0, &param);
-  if (ret < 0)
+  ret = spawn_proxyattrs(attr);
+  if (ret == OK)
     {
-      serr("ERROR: nxsched_get_param failed: %d\n", ret);
-      nxmutex_unlock(&g_spawn_parmlock);
-      return -ret;
-    }
+      /* Start the task */
 
-  /* Disable pre-emption so that the proxy does not run until waitpid
-   * is called.  This is probably unnecessary since the nxposix_spawn_proxy
-   * has the same priority as this thread; it should be schedule behind
-   * this task in the ready-to-run list.
-   */
+      ret = nxposix_spawn_exec(pid, path, actions, attr, argv, envp);
+#ifdef CONFIG_SCHED_HAVE_PARENT
```

Comment:
Is this re-parenting necessary any more ? The proxy thread obviously needs to be re-parented but now the thread is created directly from nsh, so the correct parent is nsh right ?

Reply:
Yes, you are right, re-parent should not be needed anymore